src/vcpkg/tools.cpp

```diff
@@ -765,6 +773,7 @@ namespace vcpkg
             for (auto&& candidate : candidates)
             {
                 if (!fs.exists(candidate, IgnoreErrors{})) continue;
+                if (!tool_provider.is_acceptable(candidate)) continue;
```

Comment:
The path check is cheaper then running the tool, therefore i moved the check upwards.

Reply:
This improves CMake but regresses Python detection by trying to create virtual environments for versions we don't want anyway. Introduce a new virtual for CMake to use here instead?

Reply:
I have created the new function `cheap_is_acceptable`. If you have an better name I will take it 😆